xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/metadata/LengthSolrInputDocument.java

```diff
@@ -58,6 +66,45 @@ public void setField(String name, Object value)
             this.length += ((byte[]) value).length;
         }
 
+        this.uniqueFields.remove(name);
+
         // TODO: support more type ?
     }
+
+    /**
+     * Add a value to a field if it's not already present.
```

Comment:
I guess we need to renamed `LengthSolrInputDocument` to something more generic now (`XWikiSolrInputDocument` ?).

Reply:
Yes, that would make sense. Shall I just rename it?

Reply:
That's what I would do. You are worried that some code uses it as API ?

Reply:
Looks like you've used it in https://github.com/xwiki-contrib/api-solrsecurity/blob/f5c8b1bb94f01e603b6bea59b698d3b53319fd05/src/main/java/org/xwiki/contrib/solrsecurity/internal/CustomDocumentSolrMetadataExtractor.java#L30...

I've just renamed it, but I can also rename it back if you don't agree that we break that extension.

Reply:
> Looks like you've used it in https://github.com/xwiki-contrib/api-solrsecurity/blob/f5c8b1bb94f01e603b6bea59b698d3b53319fd05/src/main/java/org/xwiki/contrib/solrsecurity/internal/CustomDocumentSolrMetadataExtractor.java#L30...

Yes, but definitely not as an API. LengthSolrInputDocument is far from being the main problem. This extension needs to be refactored a bit now that we have SolrEntityMetadataExtractor (it did not exist back then).

Reply:
Still, the extension will be broken. So you think it is okay to break that extension?

Reply:
> This extension needs to be refactored a bit now that we have SolrEntityMetadataExtractor (it did not exist back then).

Done.